### ISSUE TITLE
LinkedList

### ISSUE THREAD

**Issue by uatach** [Kerosene2000/random-wheel-rs#5]
Is there any important reason to use a LinkedList?
A HashMap or a BTreeMap look way more promising specially because there's no need to have repeated data.

**Comment by Kerollmops** [comment]
The first time I create this project I try to use a Btree or a Hash map But of you use the proba as key it's not good, yes you need to have multiple value by key in this case

``` rust
let mut rw = RandomWheel::new();
rw.push(1., 10);
rw.push(1., 20);
```

Or, yes, I can use the data (10 or 20) as key not bad I will try thanks

**Comment by uatach** [comment]
Yes, sorry, I forgot to mention that I was thinking of data as key.

**Comment by Kerollmops** [comment]
This night (GMT +1) I will try to delete linkedList and use btree or hash map.
btree I think !
Thanks for the idea!

**Comment by Kerollmops** [comment]
That's the problem, if I want to use a:
- `BTreeMap`, I need:
  - `std::cmp::Ord` for the data `T`
- `HashMap`, I need:
  - `std::cmp::Eq`, `PartialEq`, ... for the data `T`

The solution I had in mind is using a struct with the data and the `Ord trait` or `Eq trait` are based on the `probability`:

``` rust
struct for_btree {
    data: T,
    proba: f32
}
```

I think I can't force the user to use these traits for the data... Your opinion ?

**Comment by uatach** [comment]
I see, that's tricky... what if you use a BTreeSet then and a struct that wraps `T` and `f32` and implements `Ord` to sort by proba? (then `T` doesn't need `Ord`)

**Comment by Kerollmops** [comment]
Why not, I will try ok.

**Comment by uatach** [comment]
Just now I saw that `Ord` requires a lot of stuff, do you think it is too much trouble?

**Comment by Kerollmops** [comment]
I don't know I am trying to use this and it works for the moment.
0bafb67d7a0862e48642f357b0a672806c6b1331

**Comment by Kerollmops** [comment]
I think that's good to use the `standard lib` but I am fighting for remove data and return it `pop() -> Option<T>`.

**Comment by uatach** [comment]
I see what you mean, very weird... I will read the docs a bit.

**Comment by Kerollmops** [comment]
Yes thank you, because I am actually fighting the `BTreeSet::remove()` function ;)
b97ad48659055b4790a8dcf4615aacd6bdd0232a

**Comment by Kerollmops** [comment]
use `VecDeque`.
fixed 6d1ce946eac272e1fcd365c9f8793993d162753a
